delocate/delocating.py

```diff
@@ -578,6 +592,275 @@ def _make_install_name_ids_unique(
         validate_signature(lib)
 
 
+def _get_macos_min_version(dylib_path: Path) -> Iterator[tuple[str, Version]]:
+    """Get the minimum macOS version from a dylib file.
+
+    Parameters
+    ----------
+    dylib_path : Path
+        The path to the dylib file.
+
+    Yields
+    ------
+    str
+        The CPU type.
+    Version
+        The minimum macOS version.
+    """
+    if not _is_macho_file(dylib_path):
+        return
+    for header in MachO(dylib_path).headers:
+        for cmd in header.commands:
+            if cmd[0].cmd == LC_BUILD_VERSION:
+                version = cmd[1].minos
+            elif cmd[0].cmd == LC_VERSION_MIN_MACOSX:
+                version = cmd[1].version
+            else:
+                continue
+            yield (
+                CPU_TYPE_NAMES.get(header.header.cputype, "unknown"),
+                Version(f"{version >> 16 & 0xFF}.{version >> 8 & 0xFF}"),
+            )
+            break
+
+
+def _get_archs_and_version_from_wheel_name(
+    wheel_name: str,
+) -> dict[str, Version]:
+    """
+    Get the architecture and minimum macOS version from the wheel name.
+
+    Parameters
+    ----------
+    wheel_name : str
+        The name of the wheel.
+
+    Returns
+    -------
+    Dict[str, Version]
+        A dictionary containing the architecture and minimum macOS version
+        for each architecture in the wheel name.
+    """
+    platform_tag_set = parse_wheel_filename(wheel_name)[-1]
+    platform_requirements = {}
+    for platform_tag in platform_tag_set:
+        match = _PLATFORM_REGEXP.match(platform_tag.platform)
+        if match is None:
+            raise ValueError(f"Invalid platform tag: {platform_tag.platform}")
+        major, minor, arch = match.groups()
+        platform_requirements[arch] = Version(f"{major}.{minor}")
+    return platform_requirements
+
+
+def _get_problematic_libs(
+    required_version: Optional[Version],
+    version_lib_dict: Dict[Version, List[Path]],
+    arch: str,
+) -> set[tuple[Path, Version]]:
+    """
+    Filter libraries that require more modern macOS
+    version than the provided one.
+
+    Parameters
+    ----------
+    required_version : Version or None
+        The expected minimum macOS version.
+        If None, return an empty set.
+    version_lib_dict : Dict[Version, List[Path]]
+        A dictionary containing mapping from minimum macOS version to libraries
+        that require that version.
+    arch : str
+        The architecture of the libraries. For proper handle arm64 case
+
+    Returns
+    -------
+    set[tuple[Path, Version]]
+        A set of libraries that require a more modern macOS version than the
+        provided one.
+    """
+    if required_version is None:
+        return set()
+    if arch == "arm64" and required_version < Version("11.0"):
+        required_version = Version("11.0")
```

Comment:
When checking the code again, I realized that without adding this check, the error output may be a little too verbose. 

Case. 
Someone compiled `unniversal2` wheel where all `x86_64` deps have targeted `10.9` and all except one `arm64` deps have targeted 11.0. The problematic one had target 12.0. Without this fix, all `arm64` deps will be listed as problematic. With these lines added (and modified call), only single problematic one. 

Should I add tests?

Reply:
I have trouble understanding this. I do not know or remember enough about the architecture versioning history. I'll trust your knowledge on this is correct.

Adding a test might help with explaining this to me. This is also handling a edge case which should have its own test.

Reply:
Short summary. 
The first macOS that supports arm64 processors is macOS 11.0. So no arm64 binary will be compatible with macOS from line 10.x 

Also, compilers will just simple bump the deployment target to 11.0 if the requested value is lower. 

Some time ago, wheel maintainers decided that universal2 wheels could have a platform target below 11.0 if all arm64 binaries are 11.0 compatibles. 

I have added a comment with an explanation of the function, could you check?

Reply:
That sounds right. It was a good idea to add this extra check. Everything so far looks good.